src/components/JdsCalendar/Calendar.vue

```diff
@@ -0,0 +1,189 @@
+        <jds-icon  name="chevron-down" size="18px" :rotate="-90"/>
+      </div>
+    </div>
+    <div class="jds-calendar__list-of-days">
+      <div 
+        v-for="(name,index) in listOfDays" 
+        :key="index" 
+        class="jds-calendar__name-of-day"
+      >
+        {{ name }}
+      </div>
+    </div>
+    <div class="jds-calendar__days">
+      <div 
+        @click="onCalendarClick(date)" 
+        v-for="date in days" 
+        :key="date.id"
+        :class="{
+          'jds-calendar__days-day' : true,
+          'jds-calendar__days-day--selected': isSelected(date),
+          'jds-calendar__days-day--today': isToday(date),
+          'jds-calendar__days-day--holiday': isSunday(date),
+          'jds-calendar__days-day--gray': isNotThisMonth(date),
+        }"
+      >
+        {{ date.date }}
+      </div>
+    </div>
+  </div>
+</template>
+
+<script>
+import { getDays, listOfDays, listOfMonth } from '../../utils/calendar'
+import { fnDate } from '../../utils/date-input'
+import JdsIcon from '../JdsIcon'
+
+export default {
+  name: 'jds-calendar',
+  components: {
+    JdsIcon
+  },
+  model: {
+    prop: 'value',
+    event: 'change',
+  },
+  props:{
+    /**
+     * The name for calendar
+     */
+    name: {
+      type: String
+    },
+    /**
+     * Type of calendar
+     * Value is `picker | range`
+     */
+    type: {
+     type: String,
+     default: 'picker'
+    },
+    /**
+     * You can set minimal date for
+     * the earliest acceptable date
+     * The value of the min date format follows the pattern `DD/MM/YYYY`.
+     */
+    min: {
+      type: String,
+      default: '01/01/1900'
+    },
+    /**
+     * You can set maximal date for
+     * the latest acceptable date
+     * The value of the max date format follows the pattern `DD/MM/YYYY`.
+     */
+    max: {
+      type: String,
+      default: '31/12/9999'
+    },
+    /**
+     * Bound model.
+     * @name value
+     * @model
+     * The value of the input date format follows the pattern `DD/MM/YYYY`.
+     * if value not set the value will be current date
+     */
+    value: {
+      type: String,
+    },
+  },
+  data() {
+    return {
+      mValue: undefined,
+      selectedValue: null,
+      listOfDays,
+      listOfMonth
+    }
+  },
+  watch: {
+    value: {
+      handler: function (v) {
+        if(typeof v === 'undefined' || v === null || !this.isDateStringValid(v)){
+          this.mValue = new Date()
+        }else{
+          this.mValue = fnDate.parseDate(v)
+          this.selectedValue = v.split('/')
+        }
+      },
+      immediate: true,
+    }
+  },
+  computed: {
+    days () {
+      return getDays(this.mValue)
+    }
+  },
+  methods: {
+    isDateStringValid(dateString){
+      let valid = true
+      const arrayOfDate = dateString.split("/")
+      const date = new Date(arrayOfDate[1]+"/"+arrayOfDate[0]+"/"+arrayOfDate[2])
+      if(date.toString() === 'Invalid Date'){
+        valid = false
+      }
```

Comment:
Case to test:
1) Set `dateString` to `08/02/2`
2) `isDateStringValid` will return `true`, in which Date equals `Fri Feb 08 2002 00:00:00 GMT+0700 (Western Indonesia Time)`
3) Since step (2) is valid, `parseDate` is then called
4) But, `parseDate(`08/02/2`) return null.
5) Errors are thrown due to `getMonth` and `getFullYear` within template can't be called on `null` reference.

If year is expected to consist of 4 digits, `isDateStringValid` must also check whether `arrayOfDate[2]` length is number and has sufficient length.

Reply:
I have fixed at https://github.com/jabardigitalservice/jds-design-system-vue-lib/pull/52/commits/4e927177f4f67637963c47d9b365f0ceeffcf4bc